src/GENIEManager.cpp

```diff
@@ -0,0 +1,193 @@
+///////////////////////////////////////////////////////////////////////////////
+//
+// GENIEManager.cpp
+//
+// Manages GENIE input to qpixg4
+//
+// Created by: Ilker 20122021
+// Modified by: Dave Elofson 01092022
+//
+///////////////////////////////////////////////////////////////////////////////
+
+
+// Class Includes
+#include "GENIEManager.h"
+
+// Package Includes
+#include "ConfigManager.h"
+
+// ROOT Includes
+#include "TBranch.h"
+#include "TFile.h"
+#include "TTree.h"
+
+// Geant4 includes
+#include "globals.hh"
+#include "G4AutoLock.hh"
+
+// System Includes
+#include <iostream>
+#include <string>
+
+
+
+G4ThreadLocal GENIEManager * GENIEManager::instance_=0;
+
+//--------------------------------
+GENIEManager::GENIEManager()
+  : event_(0), NParticles_(0), f_(0), tree_(0), xsec_(0.),
+  dxsec_(0.), weight_(0.), prob_(0.), 
+  NParticles(0),
+  inputFile_(""), particleType_(""), genieFormat_(""), treeName_(""),
+  isInitialized(0)
+{
+    memset(pdg_, 0, Size_*sizeof(G4int));
+    memset(vtx_, 0, Size_*sizeof(G4double));
+    memset(X4_, 0, 4*Size_*sizeof(G4double));
+    memset(P4_, 0, 4*Size_*sizeof(G4double));
+    memset(polz_, 0, 3*Size_*sizeof(G4double));
+    memset(idx, 0, Size_*sizeof(G4int));
+    memset(Status, 0, Size_*sizeof(G4int));
+    memset(rescat_, 0, Size_*sizeof(G4int));
+    memset(FirstMother, 0, Size_*sizeof(G4int));
+    memset(LastMother, 0, Size_*sizeof(G4int));
+    memset(FirstDaughter, 0, Size_*sizeof(G4int));
+    memset(LastDaughter, 0, Size_*sizeof(G4int));
+
+}
+
+
+//--------------------------------
+GENIEManager::~GENIEManager() {
+}
+//--------------------------------
+GENIEManager * GENIEManager::Instance()
+{
+  if (instance_ == 0) instance_ = new GENIEManager();
+  return instance_;
+}
+
+
+
+G4int GENIEManager::Initialize() {
+  genieFormat_ = ConfigManager::GetGenieFormat();
+  inputFile_ = ConfigManager::GetInputFile(); 
+  particleType_ = ConfigManager::GetParticleType();
+ 
+  genieFormat_.toLower();
+
+  if (genieFormat_ == "rootracker" || genieFormat_ == "grootracker") {
+    treeName_ = "gRooTracker";
+  } else {
+    G4cerr << "The genie format entered: '" << genieFormat_ << "' is not currently supported." << G4endl
+           << "Please make the necessary adjustments in GENIEManager.cpp and GENIEManager.h and try again." << G4endl
+           << G4endl
+           << "Current list of supported genie formats:" << G4endl
+           << " - rootracker" << G4endl;
+    abort();
```

Comment:
Since this is a Geant4 program, you might consider using G4Exception with the "FatalException" enum value.  Not required.